.github/workflows/build.yml

```
@@ -1,35 +1,34 @@
-# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
-
-name: Build
+name: Lint Publish
 
 on:
-  pull_request:
+  push:
     branches: [main]
 
+env:
+  REPO_NAME: clamav-java
+  JAVA_VERSION: 17.0.8
+  JAVA_DISTRIBUTION: temurin
+
 jobs:
-  build:
+  # linting and unit testing is not ready yet.
+  publish:
+    #needs: [unit-test]
     runs-on: ubuntu-latest
-    
-    services:
-      clamavd:
-        image: lokori/clamav-java
-        ports:
-          - 3310:3310
-        volumes:
-          - /tmp:/tmp
-
+    permissions:
+      contents: read
+      packages: write
     steps:
-    - name: Checkout repository
-      uses: actions/checkout@v2
-    - name: Set up JDK 11
-      uses: actions/setup-java@v1
-      with:
-        java-version: 11
-    - name: Cache Maven packages
-      uses: actions/cache@v2
-      with:
-        path: ~/.m2
-        key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
-        restore-keys: ${{ runner.os }}-m2
-    - name: Build and test with Maven
-      run: mvn -B -U clean install -Djava.io.tmpdir=/tmp
+      - uses: actions/checkout@v3
```

The most recent version of `actions/checkout` is v4.1.0.
I would suggest to use the most recent v4.
```suggestion
      - uses: actions/checkout@v4
```

---

Updated the checkout action to v4